query-projector/src/lib.rs

```diff
@@ -116,6 +116,29 @@ impl QueryResults {
             &FindRel(_)    => Box::new(|| QueryResults::Rel(vec![])),
         }
     }
+
+    pub fn to_pretty(&self) -> Result<String> {
```

Comment:
@rnewman not sure you want `QueryResults` to know how to pretty-print itself in this crate.  Speak or hold your piece.  (I don't care if it's here: we already reference EDN.)

Although... we probably want to move the pretty-printing to an optional feature, since it's not something we want to ship in Firefox itself.  So maybe this will have to go in a different crate?